### PR TITLE
Fixes whole campaign not being fetched

### PR DESCRIPTION
**Pull request by j6ll**
## Proposed Changes

* When switching to the "whole campaign" it was not re-fetching data
* Whole campaign was loading 7 days by default, not the actual campaign dates
* Switch data to hourly when the campaign is less than a 3 days old
* Switch hourly labelling to include date, since it could now be multi-day

## Testing Instructions

https://github.com/user-attachments/assets/aa4e365b-8b38-4ddd-a1c6-42c4836797a2

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/my-sites/promote-post-i2/components/campaign-item-details/index.tsx [hunk 1]
```diff
@@ -875,7 +879,7 @@ export default function CampaignItemDetails( props: Props ) {
 														label={ chartSource }
 													/>
 													{ getCampaignStatsChart(
-														campaignStats?.series[ chartSource ] ?? [],
+														campaignStats?.series[ chartSource ] ?? null,
```
Comment (j6ll) [inline client/my-sites/promote-post-i2/components/campaign-item-details/index.tsx:882]: empty array can be a valid response for no data, so I intend to handle these cases differently and so intentionally passing null here

File: client/my-sites/promote-post-i2/components/campaign-stats-line-chart/index.tsx/campaign-stats-line-chart.tsx [hunk 2]
```diff
@@ -63,7 +63,7 @@ const CampaignStatsLineChart = ( { data, source, resolution }: GraphProps ) => {
 
 	const formatDate = ( date: Date, hourly: boolean ) => {
 		const options: Intl.DateTimeFormatOptions = hourly
-			? { hour: 'numeric', minute: 'numeric' }
+			? { month: 'short', day: 'numeric', hour: 'numeric', minute: 'numeric' }
 			: { month: 'short', day: 'numeric' };
 		return new Intl.DateTimeFormat( locale, options ).format( date );
 	};
```
Comment (timur987) [inline client/my-sites/promote-post-i2/components/campaign-stats-line-chart/index.tsx/campaign-stats-line-chart.tsx:69]: VS Code suggests to move `formatDate` function to `useMemo` hook callback entirely.

Also, I noticed that `useMemo`'s dependencies can start including `locale` value.

Here's the [patch.txt](https://github.com/user-attachments/files/18036414/patch.txt).

**Review by therocket-gr — APPROVED** [review]
i couldnt test everything as i dont have a lot of active campaigns (in prod or in local) 
I tested the refetching of the data when I switch date ranges and did a thorough code review
everything LGTM

**Comment by timur987** [comment]
> i couldnt test everything as i dont have a lot of active campaigns (in prod or in local)
> I tested the refetching of the data when I switch date ranges and did a thorough code review
> everything LGTM

Hey @therocket-gr,

just letting you know that I manage having prod data locally with the trick James provided in his earlier [PR](https://github.tumblr.net/Tumblr/a8c-dsp/pull/2793). Basically, you change the ID of your test campaign to `110111` and run SQL query to populate data copied from production.

Druid has data for that campaign, so you can operate with `start_date` and `end_date` values for your local campaign and test short and long campaigns.

The intention of this message is not to ask you to test again (I'm testing it right now) since you've already do CR, but to let you know about an approach how to test similar code changes in the future.

**Comment by therocket-gr** [comment]
> > i couldnt test everything as i dont have a lot of active campaigns (in prod or in local)
> > I tested the refetching of the data when I switch date ranges and did a thorough code review
> > everything LGTM
> 
> Hey @therocket-gr,
> 
> just letting you know that I manage having prod data locally with the trick James provided in his earlier [PR](https://github.tumblr.net/Tumblr/a8c-dsp/pull/2793). Basically, you change the ID of your test campaign to `110111` and run SQL query to populate data copied from production.
> 
> Druid has data for that campaign, so you can operate with `start_date` and `end_date` values for your local campaign and test short and long campaigns.
> 
> The intention of this message is not to ask you to test again (I'm testing it right now) since you've already do CR, but to let you know about an approach how to test similar code changes in the future.

thanks for the extra info :)

**Comment by j6ll** [comment]
Yeah, this is a tricky one to test, so whatever you can test plus a code review should be adequate.  The main issue here is that the data is not being re-fetched, and a couple of minor things I fixed at the same time :)

**Review by timur987 — APPROVED** [review]
Thanks @j6ll!

The code looks great and works as advertised.

Although I left a comment about a suggestion from VS Code, it's definitely not a blocker to go ahead.

File: client/my-sites/promote-post-i2/components/campaign-stats-line-chart/index.tsx/campaign-stats-line-chart.tsx (hunk 2 above)
Comment (j6ll) [reply client/my-sites/promote-post-i2/components/campaign-stats-line-chart/index.tsx/campaign-stats-line-chart.tsx:69]: Thank you timur, I've got another PR ready for review very soon.   I will move it in that PR and ship this one.

Comment (j6ll) [reply client/my-sites/promote-post-i2/components/campaign-stats-line-chart/index.tsx/campaign-stats-line-chart.tsx:69]: Added to #97075 

https://github.com/Automattic/wp-calypso/pull/97075/files#diff-aaf45e41c89183f0b29923806184e52ba5160e979fd46ccf11ce6f61e67d56d7R66-R72

https://github.com/Automattic/wp-calypso/pull/97075/files#diff-aaf45e41c89183f0b29923806184e52ba5160e979fd46ccf11ce6f61e67d56d7R184

**Pull request merged**